app/authentication/jwt_decoder.py

```diff
@@ -67,16 +162,33 @@ def decrypt_jwt_token(self, token):
                 return self.decode_signed_jwt_token(signed_token)
             else:
                 raise NoTokenException("JWT Missing")
-        except jwt.DecodeError as e:
+        except (jwt.DecodeError, InvalidTag, InternalError, ValueError) as e:
             raise InvalidTokenException(repr(e))
 
+    def __check_jwe_protected_header(self, header):
+        header = self._base64_decode(header).decode()
+        if header.count("alg") == 0:
+            raise InvalidTokenException("Missing Algorithm")
+        if header.count("RSA-OAEP") == 0:
+            raise InvalidTokenException("Invalid Algorithm")
+        if header.count("enc") == 0:
+            raise InvalidTokenException("Missing Encoding")
+        if header.count("A256GCM") == 0:
+            raise InvalidTokenException("Invalid Encoding")
+
+    def _check_iv_length(self, iv):
+        return len(iv) == 12
```

Comment:
Really these magic numbers should be named as constants, probably at the top of the class definition e.g. IV_EXPECTED_LENGTH = 12